### ISSUE TITLE
Allow tools to specify whether they want to block joystick move and joystick click separately

### ISSUE THREAD

**Issue by art0007i** [Yellow-Dog-Man/Resonite-Issues#2310]
### Is your feature request related to a problem? Please describe.

Currently in resonite we have a "UseSecondary" option on tools which will block BOTH movement and clicking. from my experience most tools do not actually need the movement part, but do require the click. this makes the tool slightly more annoying to use on for example oculus touch controllers, because you are stuck with tank controls whenever you use such a tool.

### Describe the solution you'd like

I think this should be separated into 2 options "UseJoystickMovement" and "UseSecondary", where the first one would only block movement without blocking clicking and the second one would do the opposite. enabling both would have the same behavior as UseSecondary does currently

I think also the DevTool and ProtoFluxTool (maybe others too) should not require joystick movement if this is added.

### Describe alternatives you've considered

n/a

**Comment by shadowpanther** [comment]
This was… #125

**Comment by art0007i** [comment]
ahh I forgot to check closed issues when looking for duplicates

**Comment by shadowpanther** [comment]
I still think that there's the need to separate the button from the axes on the tool system level, because just rebinding the whole secondary "control unit" requires to sacrifice some other button. Thus, I support your request.

**Comment by Nytra** [comment]
I agree, except maybe the naming should be more like "UseSecondaryAxis" and "UseSecondaryButton"

**Comment by Stellanora64** [comment]
This was froox's response to my request for this feature. It will most likely be closed like the rest of them unfortunately.

https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/125#issuecomment-2073641815

**Comment by shiftyscales** [comment]
In #2309 yesterday @Frooxius had noted:
> That's because right now, it uses the joystick, which means it can't be used for locomotion.
> 
> We can look into it not needing to use joystick, but that's separate issue from this, because even if that one is fixed, there will still be other things that might need to take control of the joystick.

However, there are still previously standing issues which had been highlighted above, e.g. Vive wands.

Specifically for the proposed option of decoupling axis (float2) and secondary (bool) inputs- he'd noted the following:

> This would make things even way more complicated on our end. Now when designing things, we have to worry about usability of the tool with that setting on and with that setting off, rather than just dealing with a single consistent system.
> 
> I think the main thing here would be going over some of the basic principles of the tools and removing need to use the Secondary on a lot of them.
> 
> The biggest problem in decoupling them is that it won't work for all controllers - e.g. Vive wands, you cannot reliably use Secondary without clicking it.

So broadly speaking the underlying issue of users wanting to be able to move while having a tool equipped could be looked at- but the more likely answer is to focus on removing the need for the secondary input from tools that don't need it as highlighted above.

I could still see some possible utility in being able to separately choose whether Axis/Bool blocks locomotion- **IF** this issue/implementation were **narrowly scoped to the `RawDataTool` component**.

E.g.:
- If the user wanted to suppress jumping but not moving/turning (tool that uses bool but not axis)
- Or suppressing moving/turning for a tool that utilizes the axis, but not bool while still allowing jumping

Seeking input from @Frooxius.

**Comment by ko-tengu** [comment]
I've commented on this in the other issues and on discord, but having it for the RawDataTool is the minimum viable option for this. Tooltips are presented as the intended way to let anyone equip items and interact with them for games or custom functions, and it's exceptionally limiting right now. The majority of VR users are unable to actually use a tooltip to play a game with due to the way it breaks controls on the most common style of controller. There's workarounds with making protoflux scripts for standard controller or the specific controllers, but that's opaque, more complex, and more tedious to create when you simply want to make a lasertag game or a sword for people to swing around.

**Comment by shiftyscales** [comment]
> the minimum viable option

I see it more as the only viable option if we go this route at all, @ko-tengu. As noted above previously:

> This would make things even way more complicated on our end. Now when designing things, we have to worry about usability of the tool with that setting on and with that setting off, rather than just dealing with a single consistent system.

If this were just isolated to just the RawDataTool- it is on the end-user to design the tool to function properly for all controller types.

For official/functional tools, we'd be more likely to go the other route specified, e.g. redesigning tools to not need secondary.

**Comment by Sharkmare** [comment]
I think the default tools behaving differently is a whole seperate conversation, btu as for RAWdatatool

UseSecondaryAction > Locks jump on controllers that use it to jump.
UseSecondaryAxis > Locks/Adjusts movement on controller that use it to move.

This imo makes sense to me.

these flags should start off and need to be enabled by the tool creator to avoid the problem of creator not knowing the flags exist and building their tools more restricting by accident (see sane defaults)

As a note id make for creators of tools...

> These settings are not static you can just enable and disable them on the fly.

Ex: if my tool has a MODE that uses the secondary axis to allow for positioning or such, i dont need to enable "UseSecondaryAxis" until I enter that mode.
Building tools to have proper context based handling should be encouraged.

Now for the seperate question of the default dev tool.

I PERSONALLY would rework the dev tool to no longer select things by default
Instead "Select" would be a mode flag you have to enable which then locks SecodaryAction
This notably would only lock jump.

**Comment by JackTheFoxOtter** [comment]
I don't think "secondary axis" is even a thing. Secondary action for tools is just a pressed state. On some controllers secondary doesn't even map to a joystick, so "secondary axis" doesn't really make sense.

**Comment by Sharkmare** [comment]
> I don't think "secondary axis" is even a thing. Secondary action for tools is just a pressed state. On some controllers secondary doesn't even map to a joystick, so "secondary axis" doesn't really make sense.

Secondary Axis is the correct term.

For index this will be the touchpad.

<img width="795" height="504" alt="Image" src="https://github.com/user-attachments/assets/d2ca31a0-833c-406c-8f9e-122c9aa8e6f6" />

**Comment by JackTheFoxOtter** [comment]
Oh interesting, okay I wasn't aware of that.
